server/api.go

```diff
@@ -351,6 +351,8 @@ func (p *Plugin) getConnected(w http.ResponseWriter, r *http.Request, userID str
 		Organization:      config.GitHubOrg,
 	}
 
+	userID := r.Header.Get("Mattermost-User-ID")
+
 	info, _ := p.getGitHubUserInfo(userID)
```

Comment:
Nit: Not sure if necessary, but to make more clear this, I would only call getGitHubUserInfo if the userID is different to the empty string. This function will be clearer in its intent, and we will do one less call to the KVStore.

Reply:
Good point on the call to KV Store. I tried implementing it that way, but the code didn't got cleaner. By increasing the indenting, it's harder to read what the code does.

Reply:
Not sure if clearer or not, but what about deferring the marshaling and writing of resp? Something like this:
```
resp := ...
defer func () {
    b, _ := json.Marshal(resp)
    w.Write(b)
}

userID := r.Header.Get("Mattermost-User-ID")
if userID == "" {
    return
}

info, _ := p.getGitHubUserInfo(userID)
if info == nil || info.Token == nil {
   return
}
...
```

Reply:
Another solution would be to declare such function inside the function and use it before the returns.
```
resp := ...
writeResp := func () {
    b, _ := json.Marshal(resp)
    w.Write(b)
}

userID := r.Header.Get("Mattermost-User-ID")
if userID == "" {
    writeResp()
    return
}

info, _ := p.getGitHubUserInfo(userID)
if info == nil || info.Token == nil {
   writeResp()
   return
}
...
```

Reply:
Let's use the helper methods added in https://github.com/mattermost/mattermost-plugin-github/pull/252 to make the code cleaner. I will alter the code when either PR get's merged.